### DISCUSSION TITLE
"Error: value should be string" vue.js

### DISCUSSION THREAD

**Discussion by CLMokbel** [uNmAnNeR/imaskjs#859] [Q&A]
I'm using the pattern mask, I have iMask imported. Here's a snippet. Vue 2. iMask v 6.6.1

`                <AIOInputLabel
                               id="interestRate"
                        :disable="isDisabledFormControl('interestRate')"
                        :icon="interestRate.icon"
                        :iconPercent="interestRate.iconPercent"
                        :isPercent="interestRate.isPercent"
                        :label="interestRate.label"
                        :placeholder="interestRate.placeholder"
                        :type="interestRate.type"
                        :value.sync="interestRate.value"
                                v-bind:class = "(isDisabledFormControl('interestRate'))?'noborder':currentClassIR"
                />`

I have type set to text,

and using simple guide code for mask pattern in my mounted() function
`        var patternMask = IMask(document.getElementById("interestRate"), {
            mask: '0[00][.][000]',
            lazy: true,
            placeholder: '0'
        });`

i get discussion title as error in the console

[Vue warn]: Error in mounted hook: "Error: value should be string"

secondary error : Error: value should be string

Does my project using webpack have anything to do with the values not computing?

**Comment by CLMokbel** [comment]
I found an area in my beforeMount() option where the value was being converted to a number, i changed this by adding the .toString() method to my value and the error persists.

**Reply by CLMokbel** [reply]
I changed my mask library to the vue-imask but i still get this error even though my value is a string. and the type is defined as text. Has anyone found anything 

I've added the v-imask directive to my code and created the mask in my data option and returned it to the component.

`        <AIOInputLabel
          :value.sync="interestRate.value"
          :placeholder="interestRate.placeholder"
          :type="interestRate.type"
          :disable="isDisabledFormControl('interestRate')"
          :label="interestRate.label"
          :icon="interestRate.icon"
          v-imask="interestMask"
          :iconPercent="interestRate.iconPercent"
          :isPercent="interestRate.isPercent"
           v-bind:class = "(isDisabledFormControl('interestRate')?'noborder': currentClassSIR)"
        />`

I've switched value to model in hopes that it would help using the directive but it doesn't.

**Reply by uNmAnNeR** [reply]
@CLMokbel i need some example to run to be able to help

**Reply by CLMokbel** [reply]
> @CLMokbel i need some example to run to be able to help

Thank you but i fixed the issue by moving the logic from the template component into the input component. There must have been an issue with the input component being defined in separate file.
